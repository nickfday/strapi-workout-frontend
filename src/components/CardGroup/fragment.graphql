fragment Cards on ComponentGlobalComponentBuilder {
  cards {
    id
    title
    body
    media {
      url
    }
    link {
      title
      url
    }
  }
}
