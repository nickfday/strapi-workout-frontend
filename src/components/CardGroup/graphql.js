import gql from 'graphql-tag';

const cardFragment = gql`
  fragment Cards on ComponentGlobalComponentBuilder {
    cards {
      id
      title
      body
      media {
        url
      }
      link {
        title
        url
      }
    }
  }
`;
export default cardFragment;
