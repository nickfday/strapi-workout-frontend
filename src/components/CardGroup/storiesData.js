const cardGroupData = {
  cardGroup: [
    {
      id: '1',
      title: 'Exercises',
      body:
        'Browse our exercise database with instructions, information and videos.',
      media: {
        url: '/uploads/istockphoto-614107160-1024x1024_e8459fa052.jpeg',
      },
      link: {
        title: 'exercises',
        url: '/exercises',
      },
    },
    {
      id: '2',
      title: 'Routines',
      body:
        'Browse our list of workout routines and find the right one for your goals.',
      media: {
        url: '/uploads/calendar-cup-depth-of-field-210660_3785bd3a87.jpeg',
      },
      link: {
        title: 'routines',
        url: '/routines',
      },
    },
  ],
};

const cardData = {
  cardGroup: [
    {
      id: '1',
      title: 'Exercises',
      body:
        'Browse our exercise database with instructions, information and videos.',
      media: {
        url: '/uploads/istockphoto-614107160-1024x1024_e8459fa052.jpeg',
      },
      link: {
        title: 'exercises',
        url: '/exercises',
      },
    },
  ],
};

export { cardGroupData, cardData };
