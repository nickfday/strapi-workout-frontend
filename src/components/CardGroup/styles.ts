import styled from 'styled-components';
import CardMedia from '@material-ui/core/CardMedia';

const StyledLinkWrapper = styled.div`
  text-align: center;
`;

const StyledCardMedia = styled(CardMedia)`
  height: 250px;
`;

export { StyledLinkWrapper, StyledCardMedia };
