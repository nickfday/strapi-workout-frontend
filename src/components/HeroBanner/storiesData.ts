const heroBannerImageData = {
  title: 'Workout',
  media: {
    url: '/uploads/man-sitting-on-edge-facing-sunset-915972_287d08a7d4.jpeg',
  },
  video: null,
};

const heroBannerVideoData = {
  title: 'Workout',
  media: null,
  video: {
    url: '/uploads/istockphoto-video_9045b6c3d4.mp4',
  },
};

export { heroBannerImageData, heroBannerVideoData };
