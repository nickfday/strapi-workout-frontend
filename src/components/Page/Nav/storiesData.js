const NavData = {
  data: {
    NavData: [
      {
        id: '1',
        title: 'Home',
        link: {
          title: 'Home',
          url: '/home',
        },
      },
      {
        id: '2',
        title: 'Exercises',
        link: {
          title: 'exercises',
          url: '/exercises',
        },
      },
      {
        id: '3',
        title: 'Routines',
        link: {
          title: 'routines',
          url: '/routines',
        },
      },
      {
        id: '4',
        title: 'Articles',
        link: {
          title: 'articles',
          url: '/articles',
        },
      },
    ],
  },
};

export default NavData;
