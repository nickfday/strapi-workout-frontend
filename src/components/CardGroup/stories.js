import React from 'react';
import { withKnobs, text, boolean, number } from '@storybook/addon-knobs';

import CardGroup from '.';

import items from './storiesData';

const data = {
  page: {
    components: [
      {
        cardGroup: [
          {
            id: '1',
            title: 'Exercises',
            body:
              'Browse our exercise database with instructions, information and videos.',
            media: {
              url: '/uploads/istockphoto-614107160-1024x1024_e8459fa052.jpeg',
              formats: {
                thumbnail: {
                  hash: 'thumbnail_istockphoto-614107160-1024x1024_e8459fa052',
                  ext: '.jpeg',
                  mime: 'image/jpeg',
                  width: 245,
                  height: 130,
                  size: 10.75,
                  url:
                    '/uploads/thumbnail_istockphoto-614107160-1024x1024_e8459fa052.jpeg',
                },
                large: {
                  hash: 'large_istockphoto-614107160-1024x1024_e8459fa052',
                  ext: '.jpeg',
                  mime: 'image/jpeg',
                  width: 1000,
                  height: 532,
                  size: 91.41,
                  url:
                    '/uploads/large_istockphoto-614107160-1024x1024_e8459fa052.jpeg',
                },
                medium: {
                  hash: 'medium_istockphoto-614107160-1024x1024_e8459fa052',
                  ext: '.jpeg',
                  mime: 'image/jpeg',
                  width: 750,
                  height: 399,
                  size: 58.94,
                  url:
                    '/uploads/medium_istockphoto-614107160-1024x1024_e8459fa052.jpeg',
                },
                small: {
                  hash: 'small_istockphoto-614107160-1024x1024_e8459fa052',
                  ext: '.jpeg',
                  mime: 'image/jpeg',
                  width: 500,
                  height: 266,
                  size: 31.49,
                  url:
                    '/uploads/small_istockphoto-614107160-1024x1024_e8459fa052.jpeg',
                },
                __typename: 'UploadFile',
              },
            },
            link: {
              title: 'exercises',
              url: '/exercises',
            },
          },
          {
            id: '2',
            title: 'Routines',
            body:
              'Browse our list of workout routines and find the right one for your goals.',
            media: {
              url: '/uploads/istockphoto-614107160-1024x1024_e8459fa052.jpeg',
              formats: {
                thumbnail: {
                  hash: 'thumbnail_istockphoto-614107160-1024x1024_e8459fa052',
                  ext: '.jpeg',
                  mime: 'image/jpeg',
                  width: 245,
                  height: 130,
                  size: 10.75,
                  url:
                    '/uploads/thumbnail_istockphoto-614107160-1024x1024_e8459fa052.jpeg',
                },
                large: {
                  hash: 'large_istockphoto-614107160-1024x1024_e8459fa052',
                  ext: '.jpeg',
                  mime: 'image/jpeg',
                  width: 1000,
                  height: 532,
                  size: 91.41,
                  url:
                    '/uploads/large_istockphoto-614107160-1024x1024_e8459fa052.jpeg',
                },
                medium: {
                  hash: 'medium_istockphoto-614107160-1024x1024_e8459fa052',
                  ext: '.jpeg',
                  mime: 'image/jpeg',
                  width: 750,
                  height: 399,
                  size: 58.94,
                  url:
                    '/uploads/medium_istockphoto-614107160-1024x1024_e8459fa052.jpeg',
                },
                small: {
                  hash: 'small_istockphoto-614107160-1024x1024_e8459fa052',
                  ext: '.jpeg',
                  mime: 'image/jpeg',
                  width: 500,
                  height: 266,
                  size: 31.49,
                  url:
                    '/uploads/small_istockphoto-614107160-1024x1024_e8459fa052.jpeg',
                },
              },
              __typename: 'UploadFile',
            },
            link: {
              title: 'routines',
              url: '/routines',
            },
          },
        ],
      },
    ],
  },
};

export default {
  title: 'Card Group',
  component: CardGroup,
  decorators: [withKnobs],
};

export const Basic = () => <CardGroup {...items} />;
