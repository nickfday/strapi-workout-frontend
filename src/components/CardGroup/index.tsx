import React from 'react';
import gql from 'graphql-tag';

import {
  Box,
  CardActionArea,
  CardContent,
  Typography,
  Grid,
} from '@material-ui/core';
import MaterialCard from '@material-ui/core/Card';
import Link, { LinkProps } from 'components/Link';
import { MediaProps } from 'types/media';
import { StyledCardMedia } from './styles';

export interface CardGroupProps {
  cardGroup: CardProps[];
}

export interface CardProps {
  title: string;
  id: number;
  body: string;
  link: LinkProps;
  media: MediaProps;
}

const CARD_FRAGMENT = gql`
  fragment cards on ComponentGlobalCard {
    cardGroup {
      title
      body
      id
      media {
        url
        formats
      }
      link {
        title
        url
      }
    }
  }
`;

const Card: React.FC<CardProps> = ({ title, body, link, media }) => {
  return (
    <MaterialCard data-testid="card">
      <Link href={link.url}>
        <CardActionArea>
          <StyledCardMedia
            image={media.formats.medium.url}
            title="Contemplative Reptile"
            data-testid="imageUrl"
          />

          <CardContent>
            <Typography
              gutterBottom
              variant="h5"
              component="h2"
              data-testid="title"
            >
              {title}
            </Typography>
            <Typography
              variant="body2"
              color="textSecondary"
              component="p"
              data-testid="body"
            >
              {body}
            </Typography>
          </CardContent>
        </CardActionArea>
      </Link>
    </MaterialCard>
  );
};

const CardGroup: React.FC<CardGroupProps> = (props) => {
  return (
    <Box mb={5}>
      <Grid container spacing={3}>
        {props.cardGroup.map((item) => {
          return (
            <Grid item xs={12} sm={6} key={item.id}>
              <Card {...item} />
            </Grid>
          );
        })}
      </Grid>
    </Box>
  );
};

export default CardGroup;
export { CARD_FRAGMENT, Card };
