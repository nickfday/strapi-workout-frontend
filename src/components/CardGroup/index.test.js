import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';

import CardGroup from '.';
import { cardGroupData, cardData } from './storiesData';

afterEach(cleanup);

describe('CardGroup', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<CardGroup {...cardGroupData} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the correct number of cards', () => {
    const { getAllByTestId } = render(<CardGroup {...cardGroupData} />);
    expect(getAllByTestId('card')).toHaveLength(2);
  });

  it('renders props correctly', () => {
    const { getByTestId } = render(<CardGroup {...cardData} />);
    expect(getByTestId('title').textContent).toBe('Exercises');
    expect(getByTestId('body').textContent).toBe(
      'Browse our exercise database with instructions, information and videos.',
    );
  });
});
