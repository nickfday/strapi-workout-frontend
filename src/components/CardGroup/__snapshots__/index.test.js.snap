// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardGroup matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          data-testid="card"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardMedia-root styles__StyledCardMedia-sc-1vbmfog-1 gKgaUH"
              data-testid="imageUrl"
              style="background-image: url(http://localhost:1337/uploads/istockphoto-614107160-1024x1024_e8459fa052.jpeg);"
              title="Contemplative Reptile"
            />
            <div
              class="MuiCardContent-root"
            >
              <h2
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
                data-testid="title"
              >
                Exercises
              </h2>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                data-testid="body"
              >
                Browse our exercise database with instructions, information and videos.
              </p>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiCardActions-root MuiCardActions-spacing"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <a
                  data-testid="anchor"
                  href="/exercises"
                >
                  exercises
                </a>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          data-testid="card"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardMedia-root styles__StyledCardMedia-sc-1vbmfog-1 gKgaUH"
              data-testid="imageUrl"
              style="background-image: url(http://localhost:1337/uploads/calendar-cup-depth-of-field-210660_3785bd3a87.jpeg);"
              title="Contemplative Reptile"
            />
            <div
              class="MuiCardContent-root"
            >
              <h2
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
                data-testid="title"
              >
                Routines
              </h2>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                data-testid="body"
              >
                Browse our list of workout routines and find the right one for your goals.
              </p>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiCardActions-root MuiCardActions-spacing"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <a
                  data-testid="anchor"
                  href="/routines"
                >
                  routines
                </a>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
